zerver/lib/types.py

```diff
@@ -56,7 +56,13 @@ class UserDisplayRecipient(TypedDict):
     is_mirror_dummy: bool
 
 
-DisplayRecipientT: TypeAlias = Union[str, List[UserDisplayRecipient]]
+class StreamDisplayRecipient(TypedDict):
+    tag: Literal["stream-display-recipient"]
+    stream_name: str
+    deactivated: bool
```

Comment:
What constraint requires us to add `deactivated` to display recipient objects? It would be kinda nice to avoid doing so, as we are generally working towards eliminating them in favor of just having the client use its own knowledge of the stream... though we'll need them in compatibility code for a long time, so it's not a near-term change.

Reply:
Earlier I wrote something like:
`obj["deactivated"] = Stream.objects.get(id=recipient.type_id).deactivated)` in `zerver/lib/message.py`but it led to an extra query to the database. I will explore alternative methods to achieve this without fetching recipient objects for display.

Reply:
Created a distinct type to distinguish it from the display recipient object, ensuring that eliminating display recipient objects won't significantly impact it. I've included fetching the `deactivated` status in the same query as obtaining the stream name for the display recipient to avoid an extra query. If this doesn't pose an issue, a separate function could be devised to retrieve stream status for stream messages. This approach would be a bit cleaner but comes with the tradeoff of increasing the cost.